### PR TITLE
Move schema generator and federation to new generator package

### PR DESCRIPTION
**Pull request by smyrick**
### :pencil: Description
Create the new package and folder `generator`. This just moves the files and the many many imports and doc references that we have. There is not logic change.

### :link: Related Issues
https://github.com/ExpediaGroup/graphql-kotlin/discussions/1000

### CODE REVIEW

File: docs/schema-generator/customizing-schemas/directives.md [hunk 1]
```diff
@@ -112,7 +112,7 @@ class LowercaseSchemaDirectiveWiring : KotlinSchemaDirectiveWiring {
 
 While you can manually apply all the runtime wirings to the corresponding GraphQL types directly in
 `SchemaGeneratorHooks#onRewireGraphQLType`, we recommend the usage of our
-[KotlinDirectiveWiringFactory](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/directives/KotlinDirectiveWiringFactory.kt)
+[KotlinDirectiveWiringFactory](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/directives/KotlinDirectiveWiringFactory.kt)
```
Comment (dariuszkuc) [inline docs/schema-generator/customizing-schemas/directives.md:115]:
```suggestion
[KotlinDirectiveWiringFactory](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/directives/KotlinDirectiveWiringFactory.kt)
```

**Review by dariuszkuc — CHANGES_REQUESTED** [review]
While you updated module package directory in the markdown files - you did not update the target package structure (e.g. missing `generator`)

File: docs/schema-generator/customizing-schemas/generator-config.md [hunk 2]
```diff
@@ -6,9 +6,9 @@ title: Generator Configuration
 `graphql-kotlin-schema-generator` provides a single function, `toSchema,` to generate a schema from Kotlin objects. This
 function accepts four arguments: config, queries, mutations and subscriptions. The queries, mutations and subscriptions
 are a list of
-[TopLevelObjects](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/TopLevelObject.kt)
+[TopLevelObjects](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/TopLevelObject.kt)
```
Comment (dariuszkuc) [inline docs/schema-generator/customizing-schemas/generator-config.md:9]:
```suggestion
[TopLevelObjects](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/TopLevelObject.kt)
```

also link to config below needs to be updated

File: docs/schema-generator/execution/async-models.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ programming models.
 ## Coroutines
 
 `graphql-kotlin-schema-generator` has built-in support for Kotlin coroutines. Provided default
-[FunctionDataFetcher](https://github.com/ExpediaDotCom/graphql-kotlin/blob/master/graphql-kotlin-schema-generator/src/main/kotlin/com/expedia/graphql/execution/FunctionDataFetcher.kt)
+[FunctionDataFetcher](https://github.com/ExpediaDotCom/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expedia/graphql/execution/FunctionDataFetcher.kt)
```
Comment (dariuszkuc) [inline docs/schema-generator/execution/async-models.md:13]:
```suggestion
[FunctionDataFetcher](https://github.com/ExpediaDotCom/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expedia/graphql/generator/execution/FunctionDataFetcher.kt)
```

File: docs/schema-generator/execution/fetching-data.md [hunk 4]
```diff
@@ -51,7 +51,7 @@ expose all your GraphQL fields that require some additional computations through
 ### Customizing Default Behavior
 
 You can provide your own custom data fetchers to resolve functions and properties by providing an instance of
-[KotlinDataFetcherFactoryProvider](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/execution/KotlinDataFetcherFactoryProvider.kt#L31)
-to your [SchemaGeneratorConfig](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/SchemaGeneratorConfig.kt).
+[KotlinDataFetcherFactoryProvider](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/KotlinDataFetcherFactoryProvider.kt#L31)
+to your [SchemaGeneratorConfig](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/SchemaGeneratorConfig.kt).
```
Comment (dariuszkuc) [inline docs/schema-generator/execution/fetching-data.md:55]:
```suggestion
to your [SchemaGeneratorConfig](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/SchemaGeneratorConfig.kt).
```

File: docs/schema-generator/schema-generator-getting-started.md [hunk 5]
```diff
@@ -86,7 +86,7 @@ below on why we use this wrapper class. The `config` contains all the extra info
 custom hooks, supported packages, and name overrides. See the [Generator Configuration](../customizing-schemas/generator-config) documentation for more information.
 
 You can see the definition for `toSchema` [in the
-source](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/toSchema.kt)
+source](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/executionoSchema.kt)
```
Comment (dariuszkuc) [inline docs/schema-generator/schema-generator-getting-started.md:89]:
```suggestion
source](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/toSchema.kt)
```

File: examples/client/server/src/main/kotlin/com/expediagroup/graphql/examples/SimpleMutations.kt [hunk 6]
```diff
@@ -1,6 +1,6 @@
 package com.expediagroup.graphql.examples
 
-import com.expediagroup.graphql.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
```
Comment (dariuszkuc) [inline examples/client/server/src/main/kotlin/com/expediagroup/graphql/examples/SimpleMutations.kt:3]: import order

File: examples/client/server/src/main/kotlin/com/expediagroup/graphql/examples/SimpleQueries.kt [hunk 7]
```diff
@@ -1,6 +1,6 @@
 package com.expediagroup.graphql.examples
 
-import com.expediagroup.graphql.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
```
Comment (dariuszkuc) [inline examples/client/server/src/main/kotlin/com/expediagroup/graphql/examples/SimpleQueries.kt:3]: import order

File: examples/federation/base-app/src/main/kotlin/com/expediagroup/graphql/examples/query/WidgetQuery.kt [hunk 8]
```diff
@@ -16,7 +16,7 @@
 
 package com.expediagroup.graphql.examples.query
 
-import com.expediagroup.graphql.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
```
Comment (dariuszkuc) [inline examples/federation/base-app/src/main/kotlin/com/expediagroup/graphql/examples/query/WidgetQuery.kt:19]: looks like `examples.model` below is out of order

File: examples/server/ktor-server/src/main/kotlin/com/expediagroup/graphql/examples/ktor/ktorGraphQLSchema.kt [hunk 9]
```diff
@@ -16,14 +16,14 @@
...
-import com.expediagroup.graphql.TopLevelObject
+import com.expediagroup.graphql.generator.SchemaGeneratorConfig
+import com.expediagroup.graphql.generator.TopLevelObject
 import com.expediagroup.graphql.examples.ktor.schema.BookQueryService
 import com.expediagroup.graphql.examples.ktor.schema.CourseQueryService
 import com.expediagroup.graphql.examples.ktor.schema.HelloQueryService
 import com.expediagroup.graphql.examples.ktor.schema.LoginMutationService
 import com.expediagroup.graphql.examples.ktor.schema.UniversityQueryService
-import com.expediagroup.graphql.toSchema
+import com.expediagroup.graphql.generator.toSchema
```
Comment (dariuszkuc) [inline examples/server/ktor-server/src/main/kotlin/com/expediagroup/graphql/examples/ktor/ktorGraphQLSchema.kt:26]: import order

File: examples/server/ktor-server/src/main/kotlin/com/expediagroup/graphql/examples/ktor/schema/BookQueryService.kt [hunk 10]
```diff
@@ -15,7 +15,7 @@
  */
 package com.expediagroup.graphql.examples.ktor.schema
 
-import com.expediagroup.graphql.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
```
Comment (dariuszkuc) [inline examples/server/ktor-server/src/main/kotlin/com/expediagroup/graphql/examples/ktor/schema/BookQueryService.kt:18]: import order

File: examples/server/spring-server/src/main/kotlin/com/expediagroup/graphql/examples/Application.kt [hunk 11]
```diff
@@ -16,7 +16,7 @@
 
 package com.expediagroup.graphql.examples
 
-import com.expediagroup.graphql.directives.KotlinDirectiveWiringFactory
+import com.expediagroup.graphql.generator.directives.KotlinDirectiveWiringFactory
```
Comment (dariuszkuc) [inline examples/server/spring-server/src/main/kotlin/com/expediagroup/graphql/examples/Application.kt:19]: import order

File: examples/server/spring-server/src/main/kotlin/com/expediagroup/graphql/examples/mutation/WidgetMutation.kt [hunk 12]
```diff
@@ -16,7 +16,7 @@
 
 package com.expediagroup.graphql.examples.mutation
 
-import com.expediagroup.graphql.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
```
Comment (dariuszkuc) [inline examples/server/spring-server/src/main/kotlin/com/expediagroup/graphql/examples/mutation/WidgetMutation.kt:19]: import order

File: examples/server/spring-server/src/main/kotlin/com/expediagroup/graphql/examples/query/CustomDirectiveQuery.kt [hunk 13]
```diff
@@ -16,7 +16,7 @@
 
 package com.expediagroup.graphql.examples.query
 
-import com.expediagroup.graphql.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
```
Comment (dariuszkuc) [inline examples/server/spring-server/src/main/kotlin/com/expediagroup/graphql/examples/query/CustomDirectiveQuery.kt:19]: import order

File: generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/toFederatedSchema.kt [hunk 14]
```diff
@@ -32,12 +32,12 @@ import graphql.schema.GraphQLSchema
  */
 @Throws(GraphQLKotlinException::class)
 fun toFederatedSchema(
-    config: FederatedSchemaGeneratorConfig,
+    config: com.expediagroup.graphql.generator.federation.FederatedSchemaGeneratorConfig,
```
Comment (dariuszkuc) [inline generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/toFederatedSchema.kt:35]: can we just import those instead?

File: generator/graphql-kotlin-federation/src/test/kotlin/com/expediagroup/graphql/generator/federation/data/TestSchema.kt [hunk 15]
```diff
@@ -14,22 +14,20 @@
...
 import graphql.schema.GraphQLSchema
 
 internal fun federatedTestSchema(
     queries: List<TopLevelObject> = emptyList(),
     federatedTypeResolvers: List<FederatedTypeResolver<*>> = emptyList()
 ): GraphQLSchema {
-    val config = FederatedSchemaGeneratorConfig(
-        supportedPackages = listOf("com.expediagroup.graphql.federation.data.queries.federated"),
-        hooks = FederatedSchemaGeneratorHooks(federatedTypeResolvers)
+    val config = com.expediagroup.graphql.generator.federation.FederatedSchemaGeneratorConfig(
```
Comment (dariuszkuc) [inline generator/graphql-kotlin-federation/src/test/kotlin/com/expediagroup/graphql/generator/federation/data/TestSchema.kt:28]: can we just import?

File: generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/toFederatedSchema.kt (hunk 14 above)
Comment (smyrick) [reply generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/toFederatedSchema.kt:35]: Yep, IntelliJ didn't autofix I suppose

File: generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/FederatedSchemaGeneratorHooks.kt [hunk 16]
```diff
@@ -107,7 +107,7 @@ open class FederatedSchemaGeneratorHooks(private val resolvers: List<FederatedTy
      *
      * Add federated _service query to ensure it is a valid GraphQL schema.
      */
-    override fun didGenerateQueryObject(query: GraphQLObjectType): GraphQLObjectType = GraphQLObjectType.newObject(query)
+    override fun didGenerateQueryObject(type: GraphQLObjectType): GraphQLObjectType = GraphQLObjectType.newObject(type)
```
Comment (smyrick) [inline generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/FederatedSchemaGeneratorHooks.kt:110]: Matches parameter name in parent class

**Review by dariuszkuc — APPROVED** [review]

**Pull request merged**